tools/engine_tool/test/external_tools/gn_test.dart

```diff
@@ -126,4 +126,41 @@ void main() {
       isTrue,
     );
   });
+
+  test('parses a dart_test action as an executable', () async {
```

Comment:
I think this test is the best documentation of what keys should exist in a test action.  No change requested in this PR but you might consider where these keys should be documented as the complexity grows.